File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/controller/HealthController.java
package com.vipamp.vipclaw.admin.controller;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

/**
 * 健康检查控制器
 */
@RestController
@RequestMapping("/health")
@Tag(name = "健康检查", description = "系统健康检查相关接口")
public class HealthController {

    @GetMapping
    @Operation(summary = "健康检查", description = "检查系统运行状态")
    public Map<String, Object> healthCheck() {
        Map<String, Object> response = new HashMap<>();
        response.put("status", "UP");
        response.put("message", "VIPClaw Admin Service is running");
        response.put("timestamp", System.currentTimeMillis());
        return response;
    }
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/controller/HealthController.kt
package com.vipamp.vipclaw.admin.controller

import com.vipamp.vipclaw.admin.common.SystemInfo
import com.vipamp.vipclaw.admin.config.EditionUtil
import com.vipamp.vipclaw.admin.dto.ResultVo
import com.vipamp.vipclaw.admin.dto.ResultVo.Companion.success
import io.swagger.v3.oas.annotations.media.Schema
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @Author: heqingsong
 * @Date: 2026/4/19
 * @Description: Health check controller
 * @Project: vipclaw
 */
@RestController
@RequestMapping("/api/")
@Tag(name = "Health Check", description = "Health monitoring")
class HealthController(private val editionUtil: EditionUtil) {

    @GetMapping("/health")
    @Schema(description = "Health check")
    fun health(): ResultVo<String?> = success("OK")

    @GetMapping("/info")
    @Schema(description = "Get version information")
    fun version(): ResultVo<SystemInfo> = success(SystemInfo.from(editionUtil))
}
